libmamba/src/core/run.cpp

```diff
@@ -283,6 +285,39 @@ namespace mamba
     }
 #endif
 
+    // Wait for a process to finish
+    // Similar to reproc::process.wait, but also handles suspension
+    int wait_for_process(pid_t pid)
+    {
+        int status;
+        pid_t wpid;
+        while (1)
+        {
+            wpid = waitpid(pid, &status, WUNTRACED);
+            if (wpid == pid && WIFSTOPPED(status))
+            {
+                // If the child was suspended (e.g. CTRL-Z),
+                // suspend entire process group as well
+                kill(0, SIGSTOP);
+            }
+            else
+            {
+                break;
+            }
+        }
+
+        // Compute the return status mimicking the private logic in reproc++
```

Comment:
Can you add a link to that logic as a comment please

Reply:
I added a link to the main part of the logic, which is [here](https://github.com/DaanDeMeyer/reproc/blob/1c07bdbec3f2e/reproc/src/process.posix.c#L455). Do you want a second link for the `wpid < 1` part? That would be [here](https://github.com/DaanDeMeyer/reproc/blob/1c07bdbec3f2ecba7125b9499b9a8a77bf9aa8c7/reproc/src/process.posix.c#L471)